dev/lint-requirements.txt

```diff
@@ -1,5 +1,6 @@
 pyarrow==0.17.0
-pylint==2.11.1
+pylint==2.1.0
+astroid==2.7.3
```

Comment:
Do we need to update these? Downgrading pylint may require reformatting portions of the codebase. pinning astroid doesn't seem strictly essential for linting.